### PR TITLE
Adds support to display options with conditions

### PR DESCRIPTION
**Pull request by cgranleese-r7**
This PR adds support to the appropriate `show` commands to include options with conditions. This PR includes test for this new output as well.

Commands updated:
- `show options`
- `show advanced`
- `show evasion`

## Note
There are currently no `register_evasion_options` making use of conditions, so I added one in manually to test. See image below

### `show options`
image

### `show advanced`
image

### `show evasion`
image

## Verification

List the steps needed to make sure this thing works

- [ ] Start `msfconsole`
- [ ] `use ldap_query`
- [ ] **Verify** that the conditional options are being output for `show options`
- [ ] **Verify** that the conditional options are being output for `show advanced`
- [ ] **Optional** verify that the conditional options are being output for `show evasions` (will require manually adding this)

### CODE REVIEW

File: lib/msf/core/opt_condition.rb [hunk 1]
```diff
@@ -16,6 +19,36 @@ def self.eval_condition(left_value, operator, right_value)
...
+      when :nin
+        if expects_blank_values
+          return "#{left_source} is not blank"
+        end
+
+        "#{left_source} not in #{right_value.join(',')}"
+      else
+        'placeholder'
+      end
+    end
```
Comment (adfoster-r7) [inline lib/msf/core/opt_condition.rb:50]: We need to apply the feedback from here:

https://github.com/rapid7/metasploit-framework/pull/17505/files#r1082926728

https://github.com/rapid7/metasploit-framework/pull/17505/files#r1082930108

https://github.com/rapid7/metasploit-framework/pull/17505/files#r1082930627

Comment (cgranleese-r7) [reply lib/msf/core/opt_condition.rb:52]: Got all of those addressed

File: lib/msf/core/exploit/remote/kerberos/service_authenticator/options.rb [hunk 2]
```diff
@@ -11,6 +11,8 @@ module Msf::Exploit::Remote::Kerberos::ServiceAuthenticator::Options
   # @param [Array<String>] auth_methods The allowed auth methods
   # @see Msf::Exploit::Remote::Kerberos::ServiceAuthenticator::Options::Msf::Exploit::Remote::AuthOption
   def kerberos_auth_options(protocol:, auth_methods:)
+    option_conditions = ["#{protocol}::Auth", '==', 'kerberos']
```
[the remark this answers is not in the document]
Comment (cgranleese-r7) [reply lib/msf/core/exploit/remote/kerberos/service_authenticator/options.rb:14]: Got that updated.

**Comment by adfoster-r7** [comment]
## Release Notes

Updates the `show options` and `show advanced` command to visually group options with the same conditions together, such as options that require an action or datastore value to be set

**Comment by jmartin-r7** [comment]
An idea, It might be nice to have an option to request the previous behaviour at least for items based on `action` selection, as a future enhancement. Possibly consider a feature flag that sets to old behaviour and offers a `-v` flag to the `show` commands to express all possible combination requirements.

**Pull request merged**